lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -5453,10 +5459,17 @@ ParseResult FIRCircuitParser::parseCircuit(
   // proactively touch it to make sure that it is always already created.
   (void)getLexer().translateLocation(info.getFIRLoc());
 
+  // Pre-verify symbol table, so we can construct it next.  Ideally, we would do
+  // this verification through the trait.
+  if (failed(mlir::detail::verifySymbolTable(circuit)))
```

Comment:
Little nervous about verifying symbol /users/ generically at this point, but probably it's fine.

It appears SymbolTable::lookup is actually "const", can we make the plumbed reference const as well? Might help ensure accidental insertion or mutation, hopefully that "just works"?

Reply:
users shouldn't even exist, but it is an extra scan.